magni_bringup/scripts/launch_core.py

```diff
@@ -68,10 +70,26 @@ def get_conf():
         print "Error calling pifi"
         subprocess.call(["chronyc", "waitsync", "6"]) # Wait up to 60 seconds for chrony
 
+    boardRev = 0
+
+    if conf['motor_controller']['board_version'] == None: 
+        # Code to read board version from I2C
+        # The I2C chip is only present on 5.0 and newer boards
+        try:
+            i2cbus = smbus.SMBus(1)
+            i2cbus.write_byte(0x20, 0xFF)
+            time.sleep(0.2)
+            inputPortBits = i2cbus.read_byte(0x20)
```

Comment:
We need to test this for a couple cases and I can do this even if it duplicates your tests.
1) Obviously test with a rev 4.9 board so we can error out and not kill motor node
2) Test with no MCB board for same reason, to be sure we don't bomb if there is NOTHING o I2C at all (yes, I know we have other grief like RTC for this but still we must not make a new bug case
3) An even more obvious test is we verify on rev 5.x units that we set the rev to 50

Reply:
I did tests 2 and 3, I will try out test 1.